tests/test_backservice.py

```diff
@@ -0,0 +1,166 @@
+            response = self.app.post(f"{BASE_URL}/candidate?{ARGUMENTS_CREATE}")
+            self.assertEqual(response.status_code, 200)                
+            response = self.app.get(f"{BASE_URL}/candidate?{ARGUMENTS_CREATE}")
+            self.assertEqual(response.status_code, 200)
+            self.assertIn(EXPECTED_DATA_CREATE, response.json['data'])
+
+            response = self.app.post(f"{BASE_URL}/candidate?{ARGUMENTS_CREATE}")
+            self.assertEqual(response.status_code, 404)
+            self.assertIn('duplicate key value violates', response.json['msg'])
+        finally:
+            response = self.app.delete(f"{BASE_URL}/candidate?{ARGUMENTS_CREATE}")
+            self.failIf(response.status_code != 200, "Could not delete record from DB in order to do clean up of test")
+        
+    def test_post_update_valid(self):
+        try:
+            response = self.app.post(f"{BASE_URL}/candidate?{ARGUMENTS_CREATE}")
+            self.assertEqual(response.status_code, 200)        
+            response = self.app.get(f"{BASE_URL}/candidate?{ARGUMENTS_CREATE}")
+            self.assertEqual(response.status_code, 200)
+            self.assertIn(EXPECTED_DATA_CREATE, response.json['data'])
+
+            response = self.app.post(f"{BASE_URL}/candidate?{ARGUMENTS_UPDATE}")
+            response = self.app.get(f"{BASE_URL}/candidate?{ARGUMENTS_UPDATE}")
+            self.assertEqual(response.status_code, 200)
+            self.assertIn(EXPECTED_DATA_UPDATE, response.json['data'])
+        finally:
+            response = self.app.delete(f"{BASE_URL}/candidate?{ARGUMENTS_UPDATE}")
+            self.failIf(response.status_code != 200, "Could not delete record from DB in order to do clean up of test")
+        
+    def test_delete_valid(self):
+        try:
+            response = self.app.post(f"{BASE_URL}/candidate?{ARGUMENTS_CREATE}")
+            self.assertEqual(response.status_code, 200)                
+            response = self.app.get(f"{BASE_URL}/candidate?{ARGUMENTS_CREATE}")
+            self.assertEqual(response.status_code, 200)
+            self.assertIn(EXPECTED_DATA_CREATE, response.json['data'])
+        finally:
+            response = self.app.delete(f"{BASE_URL}/candidate?{ARGUMENTS_CREATE}")
+            self.assertEqual(response.status_code, 200)
+            self.assertIn("no data could be returned", response.json['msg'])
+
+        response = self.app.get(f"{BASE_URL}/candidate?{ARGUMENTS_CREATE}")
+        self.assertEqual(response.status_code, 200)
+        self.assertIn("no data could be returned", response.json['msg'])
+
+    def test_get_req_params_not_valid(self):  
+        response = self.app.get(f"{BASE_URL}/candidate?key=1111111111")
+        self.assertEqual(response.status_code, 404)
+        self.assertIn("argument is not exist", response.json['msg'])
+
+    def test_post_req_params_not_valid(self):
+        response = self.app.post(f"{BASE_URL}/candidate?key=1111111111&action=create")
```

Comment:
For post request, we need to provide the action argument so we will know if the request is an update/create.
If we do not give this argument the request will not try to continue with the request because this argument is missing,
Therefore I needed to give the argument with the 'key' argument (valid request will be with 'id' argument)